src/RuleSet/DeclarationBlock.php

```diff
@@ -40,19 +40,48 @@ public static function parse(ParserState $parserState, ?CSSList $list = null): ?
         $comments = [];
         $result = new DeclarationBlock($parserState->currentLine());
         try {
+            $selectors = [];
             $selectorParts = [];
+            $functionNestingLevel = 0;
             do {
                 $selectorParts[] = $parserState->consume(1)
-                    . $parserState->consumeUntil(['{', '}', '\'', '"'], false, false, $comments);
-                if (\in_array($parserState->peek(), ['\'', '"'], true) && \substr(\end($selectorParts), -1) != '\\') {
-                    if (!isset($stringWrapperCharacter)) {
-                        $stringWrapperCharacter = $parserState->peek();
-                    } elseif ($stringWrapperCharacter === $parserState->peek()) {
-                        unset($stringWrapperCharacter);
-                    }
+                    . $parserState->consumeUntil(['{', '}', '\'', '"', '(', ')', ','], false, false, $comments);
+                $nextCharacter = $parserState->peek();
+                switch ($nextCharacter) {
+                    case '\'':
+                    case '"':
+                        if (!isset($stringWrapperCharacter)) {
+                            $stringWrapperCharacter = $nextCharacter;
+                        } elseif ($stringWrapperCharacter === $nextCharacter) {
+                            if (\substr(\end($selectorParts), -1) !== '\\') {
+                                unset($stringWrapperCharacter);
+                            }
+                        }
+                        break;
+                    case '(':
+                        if (!isset($stringWrapperCharacter)) {
+                            ++$functionNestingLevel;
+                        }
+                        break;
+                    case ')':
+                        if (!isset($stringWrapperCharacter)) {
+                            if ($functionNestingLevel <= 0) {
+                                throw new UnexpectedTokenException('anything but', ')');
+                            }
+                            --$functionNestingLevel;
+                        }
+                        break;
+                    case ',':
+                        if (!isset($stringWrapperCharacter) && $functionNestingLevel === 0) {
+                            $selectors[] = \implode('', $selectorParts);
+                            $selectorParts = [];
+                            $parserState->consume(1);
+                        }
+                        break;
                 }
-            } while (!\in_array($parserState->peek(), ['{', '}'], true) || isset($stringWrapperCharacter));
-            $result->setSelectors(\implode('', $selectorParts), $list);
+            } while (!\in_array($nextCharacter, ['{', '}'], true) || isset($stringWrapperCharacter));
+            $selectors[] = \implode('', $selectorParts); // add final or only selector
+            $result->setSelectors($selectors, $list);
```

Comment:
This cleanup should also go into a separate pre-patch.

Reply:
That's not really plausible.  Until there is splitting at commas, `$selectorParts` will just be some strings to concatenate.  I guess it's put into an array for later concatenation as an optimization.